src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -1009,6 +1012,64 @@ private static bool TryReplaceWithAlias(this ExpressionSyntax node, SemanticMode
                 return false;
             }
 
+            // Avoid the TryReplaceWithAlias algorithm if the tree has no using alias directives. Since the input node
```

Comment:
Nitpick: should we move this whole addition to a separate function? We're making an already long method 60%ish longer, and then that means "originalModel" doesn't leak to the rest of the function where it might accidentally be used.

Reply:
➡️ I started with extracting the entire addition, but felt that it was more difficult to explain the use of `TryGetRoot`. I extracted just the local function `HasUsingAliasDirective` instead.